decayAmplitude/waveDescription.cc

```diff
@@ -136,41 +169,108 @@ namespace {
 	}
 
 
-	std::vector<configPtr>
+	vector<configPtr>
 	expand(const configPtr& config)
 	{
-		std::vector<configPtr> expanded;
+		vector<configPtr> expanded;
 
-		libconfig::Setting& root     = config->getRoot();
-		libconfig::Setting* toExpand = findExpand(root);
+		Setting& root     = config->getRoot();
+		Setting* toExpand = findExpand(root);
 		if (not toExpand) {
 			expanded.push_back(config);
 			return expanded;
 		}
-		libconfig::Setting& parent = toExpand->getParent();
+		Setting& parent = toExpand->getParent();
 
-		const std::string expandType = getExpandType(toExpand);
+		const string expandType = getExpandType(toExpand);
 		if (expandType == "binning") {
-			const std::string         binningName = getExpandName(toExpand);
-			const std::vector<double> binning     = getBinning(toExpand);
+			const string         binningName = getExpandName(toExpand);
+			const vector<double> binning     = getBinning(toExpand);
 
 			parent.remove(expandName);
 
 			for (size_t bin=0; bin<binning.size()-1; ++bin) {
-				parent.add(binningName,libconfig::Setting::TypeList);
-				parent[binningName.c_str()].add(libconfig::Setting::TypeFloat) = binning[bin];
-				parent[binningName.c_str()].add(libconfig::Setting::TypeFloat) = binning[bin+1];
+				parent.add(binningName,Setting::TypeList);
+				parent[binningName.c_str()].add(Setting::TypeFloat) = binning[bin];
+				parent[binningName.c_str()].add(Setting::TypeFloat) = binning[bin+1];
 
-				configPtr newConfig(new libconfig::Config);
+				configPtr newConfig(new Config);
 				copyConfig(*config, *newConfig, waveDescription::debug());
 
 				parent.remove(binningName);
 
-				const std::vector<configPtr> newExpanded = expand(newConfig);
+				const vector<configPtr> newExpanded = expand(newConfig);
+				expanded.insert(expanded.end(), newExpanded.begin(), newExpanded.end());
+			}
+		} else if (expandType == "taylor") {
+			int degree = 0;
+			if (not toExpand->lookupValue(degreeName, degree)) {
+				printErr << "no degree given. Aborting..." << endl;
+				throw;
+			}
+			parent.remove(expandName);
+			for (int i = 0; i < degree+1; ++i) {
+				parent.add(realCoeffName, Setting::TypeList);
+				parent.add(imagCoeffName, Setting::TypeList);
+				for (int j = 0; j < i; ++j) {
+					parent[realCoeffName.c_str()].add(Setting::TypeFloat) = 0.;
+					parent[imagCoeffName.c_str()].add(Setting::TypeFloat) = 0.;
+				}
+				parent[realCoeffName.c_str()].add(Setting::TypeFloat) = 1.;
+				parent[imagCoeffName.c_str()].add(Setting::TypeFloat) = 0.;
```

Comment:
Wouldn't it be more natural to store complex coefficients in a single array of either structs with real and imag members or of arrays of size 2?
applies also other polynomial reps like chebyshev.